commandhandler/aggregate/commandhandler.go

```diff
@@ -34,27 +35,69 @@ var ErrNilAggregateStore = errors.New("aggregate store is nil")
 // 5. The new events are stored in the event store.
 // 6. The events are published on the event bus after a successful store.
 type CommandHandler struct {
-	t     eh.AggregateType
-	store eh.AggregateStore
+	t         eh.AggregateType
+	store     eh.AggregateStore
+	useAtomic bool
+	rwMutex   *sync.RWMutex
+	a         map[string]*sync.Mutex
 }
 
 // NewCommandHandler creates a new CommandHandler for an aggregate type.
-func NewCommandHandler(t eh.AggregateType, store eh.AggregateStore) (*CommandHandler, error) {
+func NewCommandHandler(aggregateType eh.AggregateType, store eh.AggregateStore, opt ...Option) (*CommandHandler, error) {
 	if store == nil {
 		return nil, ErrNilAggregateStore
 	}
 
 	h := &CommandHandler{
-		t:     t,
+		t:     aggregateType,
 		store: store,
 	}
 
+	for i := range opt {
+		opt[i](h)
+	}
+
+	if h.useAtomic {
```

Comment:
der inhalt der if kann auch einfach in die optional func oder?

Reply:
✅